### DISCUSSION TITLE
[Multiapps][Transfer variables]

### DISCUSSION THREAD

**Discussion by coskrrb2002** [idaholab/moose#20888] [Q&A General] [answered]
Dear users,

I want to find a way to pass some variables between two apps.

Such as,

App1: Variable 1 solved -> transfer -> App2: Initial condition of Variable 1 -> Solve Variable 1 in app2 -> transfer -> App1: Initial conditions. -> ... -> until converge.

Is there any way to do this method?

I am looking up the multiapps and transfers.

However, I don't know how to define the solution of Variable1 to intial condition of app2.

**Comment by coskrrb2002** [comment, marked as answer]
I just found the multivariable_copy in the multiapps, and it works for transient as well.

**Comment by GiudGiud** [comment]
Hello

The transfers work for both steady and transient solves. 
What you are doing though may not always converge in the definition we use. You need both apps to have the same solution, otherwise what you transfer as an initial condition to app1, so the solution to app2, will have a non zero residual in the app1 solve, and the code will interpret this as the fixed point iteration not converging.

If you are expecting both app1 and app2 to converge to the same solution for Variable1 then this is ok. 

Guillaume

**Reply by coskrrb2002** [reply]
Thank you for your advice, 

I expect the same solution for Variable1.

Cheers
